Sources/LeafKit/LeafRenderer.swift

```diff
@@ -90,6 +90,19 @@ public final class LeafRenderer {
     public func render(path: String, context: [String: LeafData]) -> EventLoopFuture<ByteBuffer> {
         guard path.count > 0 else { return self.eventLoop.makeFailedFuture(LeafError(.noTemplateExists("(no key provided)"))) }
 
+        // If cache provides blocking load, try to get a flat AST immediately
+        if let blockingCache = cache as? BlockingLeafCache,
+           let cached = blockingCache.load(documentName: path),
+           cached.flat {
+            do {
+                let buffer = try serialize(cached, context: context)
+                return eventLoop.makeSucceededFuture(buffer)
+            } catch {
+                return eventLoop.makeFailedFuture(LeafError(.unknownError("Serializer Error")))
```

Comment:
If a `BlockingLeafCache` is the renderer's cache, try to hit it immediately without futures and get a flat AST to skip chaining futures for the typical call case.

Otherwise, continues to the normal entirely-future based logic for getting/rendering a template.